source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -527,6 +527,23 @@ void Filter::sendTrailers(ProcessorState& state, const Http::HeaderMap& trailers
   stats_.stream_msgs_sent_.inc();
 }
 
+void Filter::onNewTimeout(const uint32_t message_timeout_ms) {
+  // The new timeout has to be >=1ms and <= max_message_timeout configured in filter.
+  const uint32_t min_timeout_ms = 1;
+  const uint32_t max_timeout_ms = config_->maxMessageTimeout();
+  if (message_timeout_ms < min_timeout_ms || message_timeout_ms > max_timeout_ms) {
```

Comment:
An ext_proc server could still return multiple responses and effectively extend the timeout indefinitely, as long as each new_message_timeout is less than max_timeout_ms. To prevent this should we only allow a single response with new_message_timeout?

Reply:
shall we consider changing the behavior to not "extending" but "overriding"? that way no matter how many, the hard cap is still working(10s).

Reply:
@ristewar  Yes, this is achieved by adding a flag new_timeout_received_ in class ProcessorState.  And each time before stop-old-timer/restart-new-timer operation check the flag first.   Also, since we allow one such API per state, so this flag is maintained during each onStartProcessorCall() and onFinishProcessorCall().

@stevenzzzz We are stopping the old timer and start a new timer. This is "overriding", right?